verify.py

```diff
@@ -191,79 +165,122 @@ def __next__(self):
+#============================================================================
+# FEDORA RESOURCE CLASS
+#============================================================================
 
-            # RDF sources
+class FedoraResource(Resource):
+    '''Properties and methods for a resource in a Fedora repository.'''
+    def __init__(self, inputpath, config, logger):
+        Resource.__init__(self, inputpath, config, logger)
+        self.location = "fedora"
+        self.relpath = urlparse(self.origpath).path.rstrip('/')
+        self.fetch_headers()
+        if self.is_binary():
+            self.type = "binary"
+            self.metadata = self.origpath + '/fcr:metadata'
+            if self.external:
+                self.destpath = quote(
+                    (self.config.dir + self.relpath + '.external')
+                    )
+            else:
+                self.destpath = quote(
+                    (self.config.dir + self.relpath + '.binary')
+                    )
+            self.lookup_sha1()
+        else:
+            self.type = 'rdf'
+            self.destpath = quote(
+                (self.config.dir + self.relpath + self.config.ext)
+                )
+            response = requests.get(self.origpath, auth=self.config.auth)
+            if response.status_code == 200:
+                self.graph = Graph().parse(
+                    data=response.text, format="text/turtle"
+                    )
+    
+    def fetch_headers(self):
+        response = requests.head(url=self.origpath, auth=self.config.auth)
+        if response.status_code in [200, 307]:
+            self.headers = response.headers
+            self.ldp_type = response.links["type"]["url"]
+            if response.status_code == 200:
+                self.external = False
+            elif response.status_code == 307:
+                self.external = True
+            return True
+        else:
+            # handle other response codes appropriately here
+            return False
+    
+    def is_binary(self):
+        if self.headers:
+            if self.ldp_type == 'http://www.w3.org/ns/ldp#NonRDFSource':
+                return True
             else:
-                self.type = 'rdf'
-                self.destpath = quote((config.dir + self.relpath + config.ext))
-                response = requests.get(self.origpath, auth=config.auth)
-                if response.status_code == 200:
-                    self.graph = Graph().parse(data=response.text, format="text/turtle")
-
-        # handle binary resources on disk
-        elif (self.origpath.startswith(config.dir) and
-              self.origpath.endswith('.binary')):
-            self.location = 'local'
+                return False
+    
+    def lookup_sha1(self):
+        response = requests.get(self.metadata, auth=self.config.auth)
+        if response.status_code == 200:
+            m = search(
+                r'premis:hasMessageDigest <urn:sha1:(.+?)>', response.text
```

Comment:
Yeah, when the output is written, it's probably a tab character, so the spacing will be variable.  

As for the checksum and the hasMessageDigest predicate not being found - it should always be there, right?   If not, then fedora may have an issue?  What I'd do: where you set it to the empty string I'd probably write out an error message that it's not there and then let it fail the checksum test like it does now.   Not sure if that's the best approach, but it would suffice for now.